### PR TITLE
Add support for TeX MathML input

### PR DESCRIPTION
**Pull request by FiniteStateGit**
## Change summary

- Add MathJax 3.0 to Hugo template enabling TeX and MathML input.
- Include sample TeX input to smoketest.md

### Submission Checklist:

* [X] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [X] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [X] **Consistency** - Does the content consistently follow the [Style Guide](https://o3de.org/docs/contributing/to-docs/style-guide/quick-reference)?
* [X] **Help the user** - Does the documentation show the user something *meaningful*?

### CODE REVIEW

**Review by willihay — APPROVED** [review]

File: layouts/partials/javascript.html
Comment (sptramer) [inline layouts/partials/javascript.html:8]: Is there a reason not to use [KaTeX](https://katex.org/)? This has been the recommended JS renderer for web from both @hultonha and @jeremyong-az .

**Review by sptramer — COMMENTED** [review]
Will approve after the decision to use KaTeX vs. TeXML is final.

File: layouts/partials/javascript.html
Comment (hultonha) [reply layouts/partials/javascript.html:8]: I have no horse in this race and know nothing about KaTeX or TeXML 😅 @jeremyong-az is definitely the expert 😝 I'll defer to his judgement

**Review by hultonha — COMMENTED** [review]
Change looks great to me but same as @sptramer I'll wait to check if @jeremyong-az as any objections to this before approving.

**Review by sptramer — CHANGES_REQUESTED** [review]
Checked in with @jeremyong-az on Slack, and here's his feedback:

> Katex is far faster and renders better :)
> https://www.jeremyong.com/cpp/machine-learning/2020/10/23/cpp-neural-network-in-a-weekend/ here's an example with katex

So, would recommend changing the TexML to KaTeX. It looks like the rendering may be slightly more complicated, but there is a blog post I found about setting up proper Hugo integration: https://hyperupcall.github.io/blog/posts/render-latex-with-katex-in-hugo-blog/

**Review by sptramer — APPROVED** [review]
Mini-conversation with @FiniteStateGit behind the scenes we agreed that fast rendering with KaTeX can be delayed a while due to limitations behind the scenes in Hugo configuration. We're unlikely to have math-heavy pages that need fast rendering support for a bit so an issue for KaTeX investigation/integration will be filed for pick-up later.

**Pull request merged**
